source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -712,16 +734,10 @@ void Graph::setRenderMaterial(UiNodePtr node)
 
 void Graph::updateMaterials(mx::InputPtr input, mx::ValuePtr value)
 {
-
-    std::string renderablePaths;
+    std::string renderablePath;
     std::vector<mx::TypedElementPtr> elems;
-    std::vector<mx::NodePtr> materialNodes;
     mx::TypedElementPtr renderableElem;
     mx::findRenderableElements(_graphDoc, elems);
-    if (elems.size() > 0 && _renderer->getMaterials().size() == 0)
```

Comment:
The renderer's update materials does the required update so no need to do it here.